modules/Quiz/resources/quiz.js

```diff
@@ -0,0 +1,917 @@
+            });
+
+            embedPlayer.addJsListener( 'kdpReady', function(){
+                if (embedPlayer.autoplay) embedPlayer.autoplay = false;
+                embedPlayer.removeJsListener( 'kdpReady');
+            });
+            embedPlayer.addJsListener( 'playerPlayEnd', function(){
+                var anUnswered = _this.getUnansweredQuestNrs();
+                if (anUnswered) _this.almostDone(anUnswered);
+            });
+        },
+        getKClient: function () {
+            if (!this.kClient) {
+                this.kClient = mw.kApiGetPartnerClient(this.embedPlayer.kwidgetid);
+            }
+            return this.kClient;
+        },
+        getTemplateHTML: function (data) {
+            var defer = $.Deferred();
+            var quizStartTemplate = this.getTemplatePartialHTML("quizstart");
+
+            var $template = $(quizStartTemplate({
+                'quiz': this,
+                quizStartTemplate: quizStartTemplate
+            }));
+            return defer.resolve($template);
+        },
+        _continuePlay: function () {
+            var _this = this, embedPlayer = this.getPlayer();
+            if (!_this.isErr) {
+                if (_this.isScreenVisible())_this.removeScreen();
+                embedPlayer.enablePlayControls();
+                embedPlayer.play();
+                _this.displayBubbles();
+            }
+        },
+        _initParams: function () {
+            var _this = this;
+            $.grep($.quizParams.uiAttributes, function (e) {
+
+                if (e.key == "showTotalScore") {
+                    _this.showTotalScore = (e.value ==='true') ;
+                }
+                if (e.key == "canSkip") {
+                    _this.canSkip = (e.value ==='true') ;
+                }
+            });
+        },
+        _showWelcomeScreen: function () {
+            var _this = this;
+            _this.removeShowScreen("welcome");
+            $(".welcome").html(gM('mwe-quiz-welcome'));
+            $(".confirm-box").html(gM('mwe-quiz-plsWait'));
+
+            _this._checkCuepointsReady(function(){
+
+                $.grep($.quizParams.uiAttributes, function (e) {
+                    switch(e.key){
+                        case 'welcomeMessage':
+                            $(".welcomeMessage").html(e.value);
+                            break;
+                        case 'inVideoTip':
+                            if (e){
+                                $(".InvideoTipMessage").html(gM('mwe-quiz-invideoTip'));
+                            }
+                            break;
+                    }
+                });
+                $(".confirm-box").html(gM('mwe-quiz-continue'))
+                    .on('click', function () {
+                    _this.checkIfDone(0);
+                    });
+
+            });
+        },
+
+       _checkCuepointsReady:function(callback){
+           var intrVal;
+           intrVal = setInterval(function () {
```

Comment:
please assign setInterval to global(in this class) var and make sure that you dont setInterval that already started